include/benchmark/benchmark_api.h

```diff
@@ -230,7 +230,7 @@ BENCHMARK_UNUSED static int stream_init_anchor = InitializeStreams();
 // expression from being optimized away by the compiler. This function is
 // intended to add little to no overhead.
 // See: https://youtu.be/nXaxk27zwlk?t=2441
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__pnacl__) && !defined(EMSCRIPTEN)
```

Comment:
The `PNaCl` docs call out `asm("" ::: "memory")` as being explicitly supported, which is what `ClobberMemory` uses, so I guess their's hope that these will be supported eventually.

However for now it seems OK to disable them. However in the long run it's important that we find a way to provide a low/no overhead way of preventing optimizations; Calling `internal::UseChar` isn't sufficient. I'll reach out to friends working on NaCL to request a fix.